packages/tasks/src/local-apps.ts

```diff
@@ -39,36 +45,52 @@ export type LocalApp = {
 			 * And if not (mostly llama.cpp), snippet to copy/paste in your terminal
 			 * Support the placeholder {{GGUF_FILE}} that will be replaced by the gguf file path or the list of available files.
 			 */
-			snippet: (model: ModelData) => string | string[];
+			snippet: (model: ModelData) => Snippet | Snippet[];
 	  }
 );
 
 function isGgufModel(model: ModelData) {
 	return model.tags.includes("gguf");
 }
 
-const snippetLlamacpp = (model: ModelData): string[] => {
+const snippetLlamacpp = (model: ModelData): Snippet[] => {
+	const command = (binary: string) =>
+		[
+			"# Load and run the model:",
+			`${binary} \\`,
+			'  --hf-repo "${model.id}" \\',
+			"  --hf-file {{GGUF_FILE}} \\",
+			'  -p "You are a helpful assistant" \\',
+			"  --conversation",
+		].join("\n");
 	return [
-		`# Option 1: use llama.cpp with brew
-brew install llama.cpp
-
-# Load and run the model
-llama \\
-	--hf-repo "${model.id}" \\
-	--hf-file {{GGUF_FILE}} \\
-	-p "I believe the meaning of life is" \\
-	-n 128`,
-		`# Option 2: build llama.cpp from source with curl support
-git clone https://github.com/ggerganov/llama.cpp.git 
-cd llama.cpp
-LLAMA_CURL=1 make
-
-# Load and run the model
-./main \\
-	--hf-repo "${model.id}" \\
-	-m {{GGUF_FILE}} \\
-	-p "I believe the meaning of life is" \\
-	-n 128`,
+		{
+			title: "Install from brew",
+			setup: "brew install llama.cpp",
+			command: command("llama-cli"),
+		},
+		{
+			title: "Use pre-built binary",
+			setup: [
+				// prettier-ignore
+				"# Download pre-built binary from:",
+				"# https://github.com/ggerganov/llama.cpp/releases",
+			].join("\n"),
+			command: command("./llama-cli"),
+		},
+		{
+			title: "Build from source code",
+			setup: [
+				"# Install required packages",
+				"sudo apt install build-essential libcurl4-openssl-dev",
```

Comment:
Yeah I have to admit that this part is for quite technical user.

I included the `sudo apt` here is more to show which kind of dependencies needed to build the project. Personally I'm using Fedora, but for example, I can always search on google for "equivalent of libcurl4-openssl-dev on fedora".

For windows, it's recommended to use the 2nd option (download pre-built binary). But for now I don't know what's the best way to communicate that on UI level.

Another option is to just point the "setup" section to https://github.com/ggerganov/llama.cpp#build and user need to go there to read the documentation.

Reply:
I agree. I always favor keeping things simple, so I recommend keeping the environment set up for the end user to figure out.
My usual rationale is that if someone is building from the repo themselves, then they already know how to set up their environment and such.

So recommendation would still be to remove the steps to `Install required packages`

Reply:
On second thought, I think it's better to remove `Install required packages` part as you said, and also `-j` option in `make`.

The goal of this snippet is to indicate to user important steps. At worst, if the compiler cannot find libcurl, it will raise a compilation error and user can search on google what to do.

Removed in commit https://github.com/huggingface/huggingface.js/pull/778/commits/3d5db7d20e46b0d8e7569fbfb8023914dbb7773b

Reply:
Thank you! Resolving this thread! 🤗